### ISSUE TITLE
TypeError: domain.enter is not a function

### ISSUE THREAD

**Issue by Ponyboy47** [phillipj/gitlab-search#9]
I tried setting this up for a custom domain with the following command:

```bash
gitlab-search setup --domain "gitlab.imat.us" "MY_API_TOKEN"
```

and received the following error:

```
events.js:176
    domain.enter();
           ^

TypeError: domain.enter is not a function
    at Command.emit (events.js:176:12)
    at Command.module.exports.350.Command.parseOptions (/usr/lib/node_modules/gitlab-search/dist/index.js:5528:14)
    at Command.listener (/usr/lib/node_modules/gitlab-search/dist/index.js:5077:23)
    at emitTwo (events.js:106:13)
    at Command.emit (events.js:191:7)
    at Command.module.exports.350.Command.parseArgs (/usr/lib/node_modules/gitlab-search/dist/index.js:5449:12)
    at Command.module.exports.350.Command.parse (/usr/lib/node_modules/gitlab-search/dist/index.js:5272:21)
    at Object.935 (/usr/lib/node_modules/gitlab-search/dist/index.js:14851:9)
    at __webpack_require__ (/usr/lib/node_modules/gitlab-search/dist/index.js:22:30)
    at startup (/usr/lib/node_modules/gitlab-search/dist/index.js:36:19)
```

I was able to work around this by adding `"domain": "gitlab.imat.us"` to the .gitlabsearchrc file manually. Everything seemed to work as I would have expected after that.

**Comment by phillipj** [comment]
Thanks for reporting! 👍 

Would you mind sharing what version of Node.js you're using?

**Comment by mikebollandajw** [comment]
nodejs v8.10.0 also happening

**Comment by phillipj** [comment]
Thanks for yet another report about this! After some digging, and finding the culprit in Node.js core ([nodejs/node/lib/events.js#L154 v9.3.0](https://github.com/nodejs/node/blob/v9.3.0/lib/events.js#L154)) I stumbled upon this issue in commander.js which we use: https://github.com/tj/commander.js/issues/105.

That issue gives the impression that what the error we're seeing here in `gitlab-search` is because we've got a command line argument named `domain`. That in turn results in some lines inside Node.js' `EventEmitter` to run, thinking it's dealing with a legacy Node.js Domain instance and tries to execute its [`.enter()` method](https://nodejs.org/dist/latest-v9.x/docs/api/domain.html#domain_domain_enter).

Even though there's been changes in recent versions of Node.js which fixes this issue, I think it would be nice for end-users of `gitlab-search` to avoid this issue from the get-go.

Bottom line; we gotta rename the `domain` argument to something else.

**Comment by phillipj** [comment]
That old command line argument has now been renamed to `--api-domain` and published as v1.0.0 on npm.

Hopefully works better for older versions of Node.js, certainly worked better as far as I could tell on my machine.

Don't hesitate to shout if anyone is still having trouble using the `setup` command with a custom GitLab API server domain.
